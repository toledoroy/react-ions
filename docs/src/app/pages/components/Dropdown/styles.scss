@import '~react-ions/styles/common/colors';
@import '~react-ions/styles/common/mixins';

.external-controls {
  float: right;

  a {
    color: $primary-1;
    text-decoration: none;
  }
}

.wrapper {
  @include vertical-align('inline');
  white-space: nowrap;

  span {
    margin-left: 10px;
  }
}

.right-align-example {
  button {
    margin-right: unset;
  }  
}