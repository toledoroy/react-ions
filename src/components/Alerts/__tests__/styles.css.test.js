import { alertSystemWrapper, alertWrapper, countdownBar } from '../styles.css'

describe('Alerts::Styles', () => {
  describe('alertSystemWrapper', () => {
    it('should return styles for alerts that do not slide in from the right', () => {
      expect(alertSystemWrapper()).toMatchSnapshot()
    })

    it('should return styles for alerts that slide in from the right', () => {
      expect(alertSystemWrapper(true)).toMatchSnapshot()
    })
  })
  
  describe('alertWrapper', () => {
    it('should return styles for a success alert', () => {
      expect(alertWrapper('success')).toMatchSnapshot()
    })

    it('should return styles for a warning alert', () => {
      expect(alertWrapper('warning')).toMatchSnapshot()
    })

    it('should return styles for an info alert', () => {
      expect(alertWrapper('info')).toMatchSnapshot()
    })

    it('should return styles for a danger alert', () => {
      expect(alertWrapper('danger')).toMatchSnapshot()
    })

    it('should return styles for an alert that is closable', () => {
      expect(alertWrapper('success', true)).toMatchSnapshot()
    })

    it('should return styles for an alert that slides in from the right', () => {
      expect(alertWrapper('success', true, true)).toMatchSnapshot()
    })
  })

  describe('countdownBar', () => {
    it('should return countdown bar styles for a success alert', () => {
      expect(countdownBar('success', 1000)).toMatchSnapshot()
    })

    it('should return countdown bar styles for a warning alert', () => {
      expect(countdownBar('warning', 2000)).toMatchSnapshot()
    })

    it('should return countdown bar styles for an info alert', () => {
      expect(countdownBar('info', 3000)).toMatchSnapshot()
    })

    it('should return countdown bar styles for a danger alert', () => {
      expect(countdownBar('danger', 4000)).toMatchSnapshot()
    })
  })
})
