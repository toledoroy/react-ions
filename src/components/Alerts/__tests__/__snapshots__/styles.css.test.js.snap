// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts::Styles alertSystemWrapper should return styles for alerts that do not slide in from the right 1`] = `Object {}`;

exports[`Alerts::Styles alertSystemWrapper should return styles for alerts that slide in from the right 1`] = `
Object {
  "position": "fixed",
  "right": "15px",
  "top": "15px",
  "zIndex": 3850,
}
`;

exports[`Alerts::Styles alertWrapper should return styles for a danger alert 1`] = `
Array [
  "
  display: flex;
  padding: 7px 10px;
  border-radius: 4px;
  background-color: ",
  "#fff",
  ";
  color: ",
  "#243140",
  ";
  font-size: 13px;
  line-height: 1.5;
  font-weight: 200;
  box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.2), 0px 0px 1px 0px rgba(0, 0, 0, 0.2);
  margin-bottom: 15px;
  position: relative;
  overflow: hidden;

  ",
  "
  
  > svg {
    fill: ",
  "#E54C3B",
  ";
    margin-top: 1px;
    min-width: 17px;

    + div {
      margin-left: 10px;
      margin-right: ",
  "0",
  ";
    }
  }

  &:hover .countdown-bar {
    animation-play-state: paused;
  }
",
]
`;

exports[`Alerts::Styles alertWrapper should return styles for a success alert 1`] = `
Array [
  "
  display: flex;
  padding: 7px 10px;
  border-radius: 4px;
  background-color: ",
  "#fff",
  ";
  color: ",
  "#243140",
  ";
  font-size: 13px;
  line-height: 1.5;
  font-weight: 200;
  box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.2), 0px 0px 1px 0px rgba(0, 0, 0, 0.2);
  margin-bottom: 15px;
  position: relative;
  overflow: hidden;

  ",
  "
  
  > svg {
    fill: ",
  "#26B894",
  ";
    margin-top: 1px;
    min-width: 17px;

    + div {
      margin-left: 10px;
      margin-right: ",
  "0",
  ";
    }
  }

  &:hover .countdown-bar {
    animation-play-state: paused;
  }
",
]
`;

exports[`Alerts::Styles alertWrapper should return styles for a warning alert 1`] = `
Array [
  "
  display: flex;
  padding: 7px 10px;
  border-radius: 4px;
  background-color: ",
  "#fff",
  ";
  color: ",
  "#243140",
  ";
  font-size: 13px;
  line-height: 1.5;
  font-weight: 200;
  box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.2), 0px 0px 1px 0px rgba(0, 0, 0, 0.2);
  margin-bottom: 15px;
  position: relative;
  overflow: hidden;

  ",
  "
  
  > svg {
    fill: ",
  "#F29C1E",
  ";
    margin-top: 1px;
    min-width: 17px;

    + div {
      margin-left: 10px;
      margin-right: ",
  "0",
  ";
    }
  }

  &:hover .countdown-bar {
    animation-play-state: paused;
  }
",
]
`;

exports[`Alerts::Styles alertWrapper should return styles for an alert that is closable 1`] = `
Array [
  "
  display: flex;
  padding: 7px 10px;
  border-radius: 4px;
  background-color: ",
  "#fff",
  ";
  color: ",
  "#243140",
  ";
  font-size: 13px;
  line-height: 1.5;
  font-weight: 200;
  box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.2), 0px 0px 1px 0px rgba(0, 0, 0, 0.2);
  margin-bottom: 15px;
  position: relative;
  overflow: hidden;

  ",
  "
  
  > svg {
    fill: ",
  "#26B894",
  ";
    margin-top: 1px;
    min-width: 17px;

    + div {
      margin-left: 10px;
      margin-right: ",
  "20px",
  ";
    }
  }

  &:hover .countdown-bar {
    animation-play-state: paused;
  }
",
]
`;

exports[`Alerts::Styles alertWrapper should return styles for an alert that slides in from the right 1`] = `
Array [
  "
  display: flex;
  padding: 7px 10px;
  border-radius: 4px;
  background-color: ",
  "#fff",
  ";
  color: ",
  "#243140",
  ";
  font-size: 13px;
  line-height: 1.5;
  font-weight: 200;
  box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.2), 0px 0px 1px 0px rgba(0, 0, 0, 0.2);
  margin-bottom: 15px;
  position: relative;
  overflow: hidden;

  ",
  "
      animation: ",
  "
  ",
  Keyframes {
    "id": "sc-keyframes-dAZqjE",
    "inject": [Function],
    "name": "dAZqjE",
    "rules": Array [
      "@-webkit-keyframes dAZqjE{0%{opacity:0;-webkit-transform:translate3d(3000px,0,0);-ms-transform:translate3d(3000px,0,0);transform:translate3d(3000px,0,0);}80%{opacity:1;-webkit-transform:translate3d(-25px,0,0);-ms-transform:translate3d(-25px,0,0);transform:translate3d(-25px,0,0);}100%{-webkit-transform:translate3d(0,0,0);-ms-transform:translate3d(0,0,0);transform:translate3d(0,0,0);}}",
      "@keyframes dAZqjE{0%{opacity:0;-webkit-transform:translate3d(3000px,0,0);-ms-transform:translate3d(3000px,0,0);transform:translate3d(3000px,0,0);}80%{opacity:1;-webkit-transform:translate3d(-25px,0,0);-ms-transform:translate3d(-25px,0,0);transform:translate3d(-25px,0,0);}100%{-webkit-transform:translate3d(0,0,0);-ms-transform:translate3d(0,0,0);transform:translate3d(0,0,0);}}",
    ],
    "toString": [Function],
  },
  " .75s ease-out;
",
  ";
      animation-iteration-count: 1;
      max-width: 350px;
      margin-left: 15px;
    ",
  "
  
  > svg {
    fill: ",
  "#26B894",
  ";
    margin-top: 1px;
    min-width: 17px;

    + div {
      margin-left: 10px;
      margin-right: ",
  "20px",
  ";
    }
  }

  &:hover .countdown-bar {
    animation-play-state: paused;
  }
",
]
`;

exports[`Alerts::Styles alertWrapper should return styles for an info alert 1`] = `
Array [
  "
  display: flex;
  padding: 7px 10px;
  border-radius: 4px;
  background-color: ",
  "#fff",
  ";
  color: ",
  "#243140",
  ";
  font-size: 13px;
  line-height: 1.5;
  font-weight: 200;
  box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.2), 0px 0px 1px 0px rgba(0, 0, 0, 0.2);
  margin-bottom: 15px;
  position: relative;
  overflow: hidden;

  ",
  "
  
  > svg {
    fill: ",
  "#62CAF3",
  ";
    margin-top: 1px;
    min-width: 17px;

    + div {
      margin-left: 10px;
      margin-right: ",
  "0",
  ";
    }
  }

  &:hover .countdown-bar {
    animation-play-state: paused;
  }
",
]
`;

exports[`Alerts::Styles countdownBar should return countdown bar styles for a danger alert 1`] = `
Array [
  "
  animation: ",
  "
  ",
  Keyframes {
    "id": "sc-keyframes-xYoGr",
    "inject": [Function],
    "name": "xYoGr",
    "rules": Array [
      "@-webkit-keyframes xYoGr{0%{width:100%;}100%{width:0%;}}",
      "@keyframes xYoGr{0%{width:100%;}100%{width:0%;}}",
    ],
    "toString": [Function],
  },
  " ",
  "4",
  "s linear 1;
",
  ";
  height: 100%;
  position: absolute;
  background-color: ",
  "#E54C3B",
  "
",
]
`;

exports[`Alerts::Styles countdownBar should return countdown bar styles for a success alert 1`] = `
Array [
  "
  animation: ",
  "
  ",
  Keyframes {
    "id": "sc-keyframes-xYoGr",
    "inject": [Function],
    "name": "xYoGr",
    "rules": Array [
      "@-webkit-keyframes xYoGr{0%{width:100%;}100%{width:0%;}}",
      "@keyframes xYoGr{0%{width:100%;}100%{width:0%;}}",
    ],
    "toString": [Function],
  },
  " ",
  "1",
  "s linear 1;
",
  ";
  height: 100%;
  position: absolute;
  background-color: ",
  "#26B894",
  "
",
]
`;

exports[`Alerts::Styles countdownBar should return countdown bar styles for a warning alert 1`] = `
Array [
  "
  animation: ",
  "
  ",
  Keyframes {
    "id": "sc-keyframes-xYoGr",
    "inject": [Function],
    "name": "xYoGr",
    "rules": Array [
      "@-webkit-keyframes xYoGr{0%{width:100%;}100%{width:0%;}}",
      "@keyframes xYoGr{0%{width:100%;}100%{width:0%;}}",
    ],
    "toString": [Function],
  },
  " ",
  "2",
  "s linear 1;
",
  ";
  height: 100%;
  position: absolute;
  background-color: ",
  "#F29C1E",
  "
",
]
`;

exports[`Alerts::Styles countdownBar should return countdown bar styles for an info alert 1`] = `
Array [
  "
  animation: ",
  "
  ",
  Keyframes {
    "id": "sc-keyframes-xYoGr",
    "inject": [Function],
    "name": "xYoGr",
    "rules": Array [
      "@-webkit-keyframes xYoGr{0%{width:100%;}100%{width:0%;}}",
      "@keyframes xYoGr{0%{width:100%;}100%{width:0%;}}",
    ],
    "toString": [Function],
  },
  " ",
  "3",
  "s linear 1;
",
  ";
  height: 100%;
  position: absolute;
  background-color: ",
  "#62CAF3",
  "
",
]
`;
